dpctl/tests/elementwise/test_add.py

```diff
@@ -165,3 +166,79 @@ def __sycl_usm_array_interface__(self):
     c = Canary()
     with pytest.raises(ValueError):
         dpt.add(a, c)
+
+
+@pytest.mark.parametrize("op1_dtype", _all_dtypes)
+@pytest.mark.parametrize("op2_dtype", _all_dtypes)
+def test_add_dtype_out_keyword(op1_dtype, op2_dtype):
+    q = get_queue_or_skip()
+    skip_if_dtype_not_supported(op1_dtype, q)
+    skip_if_dtype_not_supported(op2_dtype, q)
+
+    sz = 127
+    ar1 = dpt.ones(sz, dtype=op1_dtype)
+    ar2 = dpt.ones_like(ar1, dtype=op2_dtype)
+
+    r = dpt.add(ar1, ar2)
+
+    y = dpt.zeros_like(ar1, dtype=r.dtype)
+    dpt.add(ar1, ar2, y)
+
+    assert np.array_equal(dpt.asnumpy(r), dpt.asnumpy(y))
+
+
+def test_add_errors():
+    ar1 = dpt.ones(2, dtype="float32", device="gpu")
```

Comment:
Gpu may mot be available, and this configuration must be handled. Does this test require a gpu though?